Tools/autotest/arducopter.py

```diff
@@ -3908,6 +3908,147 @@ def test_guided_local_velocity_target(self, vx, vy, vz_up, timeout=3):
+        # Check the last received message
+        if not (m.type_mask == (target_typemask | MAV_POS_TARGET_TYPE_MASK.LAST_BYTE) or m.type_mask == target_typemask):
+            raise NotAchievedException("Did not receive proper mask: expected=%u or %u, got=%u" %
+                  ((target_typemask | MAV_POS_TARGET_TYPE_MASK.LAST_BYTE), target_typemask, m.type_mask))
+
+        if afx - m.afx > 0.1:
+            raise NotAchievedException("Did not receive proper target acceleration afx: wanted=%f got=%f" %
+                                       (afx, m.afx))
+
+        if afy - m.afy > 0.1:
+            raise NotAchievedException("Did not receive proper target acceleration afy: wanted=%f got=%f" %
+                                       (afy, m.afy))
+
+        if afz_up - (-m.afz) > 0.1:
+            raise NotAchievedException("Did not receive proper target acceleration afz: wanted=%f got=%f" %
+                                       (afz_up, -m.afz))
+
+        self.progress("Received proper target acceleration commands")
+
+    def wait_for_local_velocity(self, vx, vy, vz_up, timeout=10):
+        """ Wait for local target velocity"""
+
+        # debug messages
+        self.progress("Waiting for local NED velocity target: (%f, %f, %f)" % (vx, vy, -vz_up))
+        self.progress("Setting LOCAL_POSITION_NED message rate to 10Hz")
+
+        # set position local ned message stream rate
+        self.set_message_rate_hz(mavutil.mavlink.MAVLINK_MSG_ID_LOCAL_POSITION_NED, 10)
+
+        # wait for position local ned message
+        tstart = self.get_sim_time()
+        while self.get_sim_time_cached() - tstart < timeout:
+
+            # get position target local ned message
+            m = self.mav.recv_match(type="LOCAL_POSITION_NED", blocking=True, timeout=1)
+
+            # could not be able to get a valid target local ned message within given time
+            if m is None:
+
+                # raise an error that did not receive a valid target local ned message within given time
+                raise NotAchievedException("Did not receive any position local ned message for 1 second!")
+
+            # got a valid target local ned message within given time
+            else:
+
+                # debug message
+                self.progress("Received position local position ned message: %s" % str(m))
+
+                # check if velocity values are in range
+                if vx - m.vx <= 0.1 and vy - m.vy <= 0.1 and vz_up - (-m.vz) <= 0.1:
+
+                    # get out of function
+                    self.progress("Vehicle successfully reached to target velocity!")
+                    return
+
+        # raise an exception
+        error_message = "Did not receive target velocities vx, vy, vz_up, wanted=(%f, %f, %f) got=(%f, %f, %f)"
+        error_message = error_message % (vx, vy, vz_up, m.vx, m.vy, -m.vz)
+        raise NotAchievedException(error_message)
+
+    def wait_for_local_acceleration(self, ax, ay, az_up, timeout=10):
```

Comment:
`wait_for_local_acceleration` fails at line 7987. 
Is this the right way to read ACC values from vehicle with SIM_STATE message?
This is not right or I implemented the `test_guided_local_acceleration_target` at line 3911 wrong.